src/dbt_score/rules/generic.py

```diff
@@ -47,7 +47,49 @@ def sql_has_reasonable_number_of_lines(
 def has_example_sql(model: Model) -> RuleViolation | None:
     """The documentation of a model should have an example query."""
     if model.language == "sql":
-        if "```sql" not in model.description:
+        if "```sql" not in (model.description or ""):
             return RuleViolation(
                 "The model description does not include an example SQL query."
             )
+
+@rule
+def has_uniqueness_test(model: Model) -> RuleViolation | None:
+    """Model has uniqueness test for primary key."""
+    if model.config.get("materialized") not in {"table", "incremental"}:
+        return None
```

Comment:
Would you like us to implement this with a rule filter? Maybe even the whole pk discovery can be a filter, and this filter can be also part of the public dbt-score API on par with default rules 🤔 

So people who implement their own rules but use the same constraints on filters can reuse it 💡